drivers/bluetooth: stm32 ipm: Fix mutex on shared memory

erwango opened the pull request:
ipm_busy is meant to be used as mutex on transport layer accesses.
Due to wrong configuration on rx_thread, imp_busy was a no-op.
Fix this.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/bluetooth: stm32 ipm: Fix mutex on shared memory

ipm_busy is meant to be used as mutex on transport layer accesses.
Due to wrong configuration on rx_thread, imp_busy was a no-op.
Fix this.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

joerchan commented on drivers/bluetooth/hci/ipm_stm32wb.c:
```diff
@@ -157,7 +157,7 @@ static void bt_ipm_rx_thread(void)
 		hcievt = k_fifo_get(&ipm_rx_events_fifo, K_FOREVER);
 
-		k_sem_take(&ipm_busy, K_NO_WAIT);
+		k_sem_take(&ipm_busy, K_FOREVER);
```
It looks like you have duplicate `k_sem_give` in the while loop here. I think maybe the ones in the switch statement can be removed.

erwango replied:
Thanks, actually it seems that the previous change broke the error management in this loop.
I've pushed a fix

erwango committed:
drivers/bluetooth: stm32wb: Fix loop

The goto instruction was removed when adding the RX thread.
While still working, this clearly break the clean handling of
error cases.
Re-instantiate the goto instruction.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

joerchan commented on drivers/bluetooth/hci/ipm_stm32wb.c:
```diff
@@ -168,8 +168,8 @@ static void bt_ipm_rx_thread(void)
 				       hcievt->evtserial.evt.evtcode);
-				k_sem_give(&ipm_busy);
-				break;
+				TL_MM_EvtDone(hcievt);
```
It looks like this wasn't here earlier, but it would make sense that it should be.

erwango replied:
indeed, this was an issue in the initial version.
